controllers/rabbitmq/rabbitmq_controller.go

```diff
@@ -356,6 +362,31 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (result ct
 
 	if clusterReady {
 		instance.Status.Conditions.MarkTrue(condition.DeploymentReadyCondition, condition.DeploymentReadyMessage)
+		// Let's wait DeploymentReadyCondition=True to apply the policy
+		queue := instance.Spec.QueueType
+		ismirrored := instance.Status.QueueType
+
+		if queue == "Mirrored" && *instance.Spec.Replicas > 1 {
+			if ismirrored != "Mirrored" {
+				Log.Info("ha-all policy not present. Applying.")
+				err := updateMirroredPolicy(ctx, helper, instance, r.config, true)
+				if err != nil {
+					Log.Error(err, "Could not apply policy")
+				} else {
```

Comment:
the else is not required, or is it. could just set `instance.Status.QueueType = "Mirrored"` if no error occurred

Reply:
we can probably return err and get rid of the else?

Reply:
right, missed to read that and expected we return on err :) yes I think we should return on err and set a condition for it